ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -389,13 +389,6 @@ PrjFS_Result PrjFS_WritePlaceholderFile(
         goto CleanupAndFail;
     }
     
-    // Expand the file to the desired size
-    if (ftruncate(fileno(file), fileSize))
-    {
-        result = PrjFS_Result_EIOError;
-        goto CleanupAndFail;
-    }
-    
     fclose(file);
```

Comment:
Did you validate that [`stat` and `fstat`](https://linux.die.net/man/2/fstat) will see the correct size with this change?

Reply:
> The change looks good overall, but I left a question on regarding testing out `stat` and `fstat`.
> 
> It would be good to hear @pmj's thoughts on this change as well.

I don't really seed `fstat()` being an issue - having a file descriptor implies the file has gone through `open()` so it'll be hydrated.

I'm personally not worried about `stat()` either as I've satisfied myself that the kernel’s implementation of it calls into the vnode listener with a hydrating action.

Nevertheless, it'd perhaps be worth adding a functional test for `stat()` - I don't know if there's a direct .NET wrapper for it we can easily access from the functional test runner, or if the functional tests are set up for easily adding some native code.[*] Not a dealbreaker for me if we don't, though - I suspect plenty of functional tests currently implicitly test this by using `stat()` as the backend to whatever high-level .NET stdlib functions they use.

[*] If it's currently not straightforward to use native code in functional tests, I wonder if that's something worth doing in itself, especially as macOS has some pretty odd syscalls such as [`exchangedata(2)`](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/exchangedata.2.html) that we may eventually want to test.

Reply:
> Nevertheless, it'd perhaps be worth adding a functional test for stat()

I agree.  We already have some C# code in the product that calls `stat()` and so it should be straightforward to do it in the functional tests as well.